### PR TITLE
feat(legacy-plugin-chart-pivot-table): add support for timestamp format

### PR DESCRIPTION
**Pull request by villebro**
🏆 Enhancements

This adds support for formatted dates on `pivot-table`. This is done by prefixing date values with `__timestamp:`, after which they can be identified and parsed into formattable date objects. In addition sorting is done based on the epoch value, meaning that the format won't affect the sorting order. Other changes:
- Add date format
- Move visual options to own tab (currently on `Query` pane)
- use time grain for formatting if smart formatting is selected
- make sorting of numeric values based on the parsed float value, not the text value

### AFTER:
image

### BEFORE:
image

### CODE REVIEW

File: plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js [hunk 1]
```diff
@@ -44,25 +45,37 @@ const propTypes = {
 };
 
 function PivotTable(element, props) {
-  const { data, height, columnFormats, numberFormat, numGroups, verboseMap } = props;
+  const { data, height, columnFormats, dateFormat, numberFormat, numGroups, verboseMap } = props;
 
   const { html, columns } = data;
   const container = element;
   const $container = $(element);
```
Comment (rusackas) [inline plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js:52]: jQuery!? 😭

Comment (ktmud) [reply plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js:52]: Unfortunately pivot table still uses `jQuery.dataTables`.

File: plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js [hunk 2]
```diff
@@ -44,25 +45,37 @@ const propTypes = {
...
+  const dateFormatter =
+    dateFormat === 'smart_date' ? smartDateFormatter : getTimeFormatter(dateFormat);
```
Comment (ktmud) [inline plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js:54]: `getTimeFormatter('smart_date')` should work just fine. But since we are here, can you update it to follow the same logic as the table chart? 

https://github.com/apache-superset/superset-ui/blob/d323ce86dd7000e9b65a2a7d6b861d713616701b/plugins/plugin-chart-table/src/transformProps.ts#L138-L151

It has fallbacks to d3-format in pre-defined columns and uses Time Grain for `smart_date`.

Comment (ktmud) [reply plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js:54]: If the code was hard to follow for you, I can also update this later and refactor these logics into some kind of common utilities.

File: plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js (hunk 1 above)
Comment (villebro) [reply plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js:52]: Hey, if it works it works! 😄

File: plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js (hunk 2 above)
Comment (villebro) [reply plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js:54]: Thanks for this reference! I didn't have strong opinions here aka. I was just out to make this work. I'll digest the table chart logic and see if it makes sense for me to streamline here 👍

**Comment by villebro** [comment]
Thanks for your reviews @ktmud and @rusackas ! <3

**Comment by villebro** [comment]
Oh, one more thing @ktmud and @rusackas . Currently the sorting carets are broken (displays a black box, see the screenshots). I wasn't sure how to fix this, do you guys know how to make the icon work properly?

**Comment by villebro** [comment]
Thanks @kristw , I'll update accordingly 👍

**Comment by ktmud** [comment]
> Oh, one more thing @ktmud and @rusackas . Currently the sorting carets are broken (displays a black box, see the screenshots). I wasn't sure how to fix this, do you guys know how to make the icon work properly?

I used `react-icons` in the table chart. https://github.com/apache-superset/superset-ui/blob/d323ce86dd7000e9b65a2a7d6b861d713616701b/plugins/plugin-chart-table/src/TableChart.tsx#L91-L98

In the future we should probably move the SVG icons from `superset-frontend` to `superset-ui` so they can be reused by visualizations.

File: plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js (hunk 2 above)
Comment (villebro) [reply plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js]: @ktmud I implemented a slightly modified version of your formatter that felt most suitable for this slightly simpler case. Does it make sense to you?

**Comment by villebro** [comment]
> I used `react-icons` in the table chart.
> 
> https://github.com/apache-superset/superset-ui/blob/d323ce86dd7000e9b65a2a7d6b861d713616701b/plugins/plugin-chart-table/src/TableChart.tsx#L91-L98
> 
> In the future we should probably move the SVG icons from `superset-frontend` to `superset-ui` so they can be reused by visualizations.

@ktmud I was unsure how to apply this snippet to this code, as it doesn't seem to be natively React. Perhaps it's vest if we merge this as-is and leave this fix for later?

File: plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js (hunk 2 above)
Comment (ktmud) [reply plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js]: Looks good! Thanks for the update. Do you mind adding a datetime column to [Storybook](https://github.com/apache-superset/superset-ui/blob/d631b17597483f981f5de67d2c482e8fae94dc55/packages/superset-ui-demo/storybook/stories/plugins/legacy-plugin-chart-pivot-table/Stories.tsx#L27) as well?

**Comment by ktmud** [comment]
> @ktmud I was unsure how to apply this snippet to this code, as it doesn't seem to be natively React. Perhaps it's best if we merge this as-is and leave this fix for later?

Sounds good
